coral-schema/src/main/java/com/linkedin/coral/schema/avro/MergeHiveSchemaWithAvro.java

```diff
@@ -58,7 +58,10 @@ public Schema struct(StructTypeInfo struct, Schema partner, List<Schema.Field> f
     } else {
       result = SchemaUtilities.copyRecord(SchemaUtilities.extractIfOption(partner), fieldResults);
     }
-    return shouldResultBeOptional ? SchemaUtilities.makeNullable(result) : result;
+    // While calling `makeNullable`, we should respect the option order of `partner`
```

Comment:
this comment is not added is every method call for `makeNullable(..)`.
Could you either add it everywhere or just add it in the class description and the method's javadoc?

Reply:
Thanks for reminding, added.